model/client.go

```diff
@@ -1341,6 +1349,24 @@ func (c *Client) UpdateLastViewedAt(channelId string, active bool) (*Result, *Ap
 	}
 }
 
+// ViewChannel performs all the actions related to viewing a channel. This includes marking
+// the channel and the previous one as read, marking the channel as being actively viewed.
+// ChannelId is required but may be blank to indicate no channel is being viewed.
+// PrevChannelId is optional, populate to indicate a channel switch occurred. Optionally
+// provide a non-zero Time, in Unix milliseconds, to manually set the viewing time.
+func (c *Client) ViewChannel(params ChannelView) (bool, *ResponseMetadata) {
```

Comment:
That would require another DB read to actually pull the time (since we use `LastPostAt` on the channel if a time isn't provided) but if needed I can add it